### PR TITLE
Move library code into josh-core crate

### PR DESCRIPTION
**Pull request by christian-schilling**

### CODE REVIEW

File: josh-proxy/src/bin/josh-proxy.rs [hunk 1]
```diff
@@ -511,10 +511,13 @@ async fn query_meta_repo(
...
-        .upstream
-        .get(upstream_protocol)
-        .ok_or(josh_error("no remote specified for the requested protocol"))?;
+    let upstream = if let Ok(host) = std::env::var("JOSH_META_HOST") {
+        host
+    } else {
+        serv.upstream
+            .get(upstream_protocol)
+            .ok_or(josh_error("no remote specified for the requested protocol"))?
+    };
```
Comment (LMG) [inline josh-proxy/src/bin/josh-proxy.rs:520]: those changes are unrelated aren't they?

Comment (christian-schilling) [reply josh-proxy/src/bin/josh-proxy.rs]: yep, fixed

**Review by LMG — APPROVED** [review]
